python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -422,42 +494,54 @@ def doProcessCcd(repo, calib_repo, processed_repo, visit, ccdnum):
         The output repository location on disk where ingested calibration images live.
     processed_repo: `str`
         The output repository location on disk where processed raw images live.
-    visit: `str`
-        One or more DECam visit numbers, e.g., `54321`. Multiple visits may be
-        indicated with standard Butler parsing, e.g., `54321^12345`.
-    ccdnum: `str`
-        One or more DECam CCDs (`1` through `62` are allowed). Setting
-        ccdnum='1..62' will process all of the CCDs.
+    dataId: `str`
+        Butler identifier naming the data to be processed (e.g., visit and ccdnum)
+        formatted in the usual way (e.g., 'visit=54321 ccdnum=7').
 
     Returns
     -------
     process_metadata: `PropertySet` or None
         Metadata from the ProcessCcdTask for use by ap_verify
 
     BASH EQUIVALENT:
-    $ processCcd.py repo --id visit=visit ccdnum=ccdnum
+    $ processCcd.py repo --id dataId
             --output processed_repo --calib calib_repo
             -C $OBS_DECAM_DIR/config/processCcdCpIsr.py
             -C processccd_config.py
             --config calibrate.doAstrometry=True calibrate.doPhotoCal=True
     ** to run from bash, 'processccd_config.py' must exist and contain
        all of the refObjLoader information in the code below. repo must also
-       already contain the ref_cats (this is done during doIngest).
+       already contain the refcats (this is done during doIngest).
 
     RESULT:
     processed_repo/visit populated with subdirectories containing the
     usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR).
     By default, the configuration for astrometric reference catalogs uses Gaia
     and the configuration for photometry reference catalogs uses Pan-STARRS.
     '''
+    lsst.log.configure()
+    log = lsst.log.Log.getLogger('ap.pipe.doProcessCcd')
+    dataId_items = re.split('[ +=]', dataId)
+    dataId_dict = dict(zip(dataId_items[::2], dataId_items[1::2]))
```

Comment:
I'm not sure I understand this. Is this basically pairing up the even- and odd-indexed elements of `dataId_items`?

Reply:
Also, what happens if `dataId` doesn't follow the right format?

Reply:
Yep. If `dataId` has an odd number of elements, it will just omit the last one and carry on. This seems as reasonable as anything.

Reply:
I'd prefer some kind of parse error, but as discussed in person parsing dataIds is not something we should have to do ourselves anyway. Maybe someday we can delegate it all to the Task framework...